src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb

```diff
@@ -0,0 +1,112 @@
+        Private ReadOnly _voidType As TypeSymbol
+
+        Friend Sub New(
+            containingModule As SourceModuleSymbol,
+            privateImplementationType As PrivateImplementationDetails,
+            voidType As NamedTypeSymbol
+        )
+            MyBase.New(containingModule, WellKnownMemberNames.StaticConstructorName, privateImplementationType)
+
+            _containingModule = containingModule
+            _privateImplementationType = privateImplementationType
+            _voidType = voidType
+        End Sub
+
+        Public Overrides ReadOnly Property IsSub As Boolean
+            Get
+                Return True
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ReturnsByRef As Boolean
+            Get
+                Return False
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ReturnType As TypeSymbol
+            Get
+                Return _voidType
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property HasSpecialName As Boolean
+            Get
+                Return True
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property MethodKind As MethodKind
+            Get
+                Return MethodKind.SharedConstructor
+            End Get
+        End Property
+
+        Friend Overrides Function GetBoundMethodBody(compilationState As TypeCompilationState, diagnostics As DiagnosticBag, Optional ByRef methodBodyBinder As Binder = Nothing) As BoundBlock
+            methodBodyBinder = Nothing
+
+            Dim factory As New SyntheticBoundNodeFactory(Me, Me, VisualBasicSyntaxTree.Dummy.GetRoot(), compilationState, diagnostics)
+
+            ' Initialize the payload root for for each kind of dynamic analysis instrumentation.
+            ' A payload root Is an array of arrays of per-method instrumentation payloads.
+            ' For each kind of instrumentation:
+            '
+            '     payloadRoot = New T(MaximumMethodDefIndex)() {}
+            '
+            ' where T Is the type of the payload at each instrumentation point, And MaximumMethodDefIndex Is the 
+            ' index portion of the greatest method definition token in the compilation. This guarantees that any
+            ' method can use the index portion of its own method definition token as an index into the payload array.
+
+            Dim payloadRootFields As IReadOnlyCollection(Of KeyValuePair(Of Integer, InstrumentationPayloadRootField)) = _privateImplementationType.GetInstrumentationPayloadRoots()
+            Debug.Assert(payloadRootFields.Count > 0)
+
+            Dim body As ArrayBuilder(Of BoundStatement) = ArrayBuilder(Of BoundStatement).GetInstance(2 + payloadRootFields.Count)
+
+            For Each payloadRoot As KeyValuePair(Of Integer, InstrumentationPayloadRootField) In payloadRootFields.OrderBy(Function(analysis) analysis.Key)
```

Comment:
Reconsidering this from the perspective of there being only one PrivateImplementationDetails class and two consumers, returning an ordered collection seems like a good idea.